projects/robots/gctronic/e-puck/controllers/e-puck2_server/e-puck2_server.c

```diff
@@ -0,0 +1,338 @@
+/*
+ * Copyright 1996-2018 Cyberbotics Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Description:  A simple program implementing a TCP/IP server emulating the
+ *               behavior of the e-puck2 robot, so that you can connect the
+ *               EPuckMonitor (Wi-Fi version) and e-puck2.vi LabVIEW panel on it.
+ *
+ * Author:       Olivier Michel
+ */
+
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <time.h>
+#ifdef _WIN32
+#include <winsock.h>
+#else
+#include <arpa/inet.h>  /* definition of inet_ntoa */
+#include <netdb.h>      /* definition of gethostbyname */
+#include <netinet/in.h> /* definition of struct sockaddr_in */
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h> /* definition of close */
+#endif
+
+#include <webots/accelerometer.h>
+#include <webots/distance_sensor.h>
+#include <webots/led.h>
+#include <webots/light_sensor.h>
+#include <webots/motor.h>
+#include <webots/robot.h>
+#include <webots/speaker.h>
+
+#include "play_melody.h"
+
+#define SOCKET_PORT 1000
+#define MOTOR_RATIO 0.00628
+
+static bool socket_init() {
+#ifdef _WIN32 /* initialize the socket API */
+  WSADATA info;
+  int rc = WSAStartup(MAKEWORD(1, 1), &info); /* Winsock 1.1 */
+  if (rc != 0) {
+    fprintf(stderr, "Cannot initialize Winsock.\n");
+    return false;
+  }
+#endif
+  return true;
+}
+
+static bool socket_set_non_blocking(int fd) {
+  if (fd < 0)
+    return false;
+#ifdef _WIN32
+  unsigned long mode = 1;
+  return (ioctlsocket(fd, FIONBIO, &mode) == 0) ? true : false;
+#else
+  int flags = fcntl(fd, F_GETFL, 0) | O_NONBLOCK;
+  return (fcntl(fd, F_SETFL, flags) == 0) ? true : false;
+#endif
+}
+
+static int socket_accept(int server_fd) {
+  int cfd;
+  struct sockaddr_in client;
+  struct hostent *client_info;
+#ifndef _WIN32
+  socklen_t asize;
+#else
+  int asize;
+#endif
+  asize = sizeof(struct sockaddr_in);
+  cfd = accept(server_fd, (struct sockaddr *)&client, &asize);
+  if (cfd == -1) {
+#ifdef _WIN32
+    int e = WSAGetLastError();
+    if (e == WSAEWOULDBLOCK)
```

Comment:
It doesn't seem useful to store this value in a variable.

Reply:
It seems you have fixed this in 51ad2edd4fcfbac26cd393e5350f4355defa8130 and the re-introduced it in 4fa865ad39ccfe46c471e7506df06e474bbdb63a